source/Politics.cpp

```diff
@@ -222,7 +222,21 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 			if((fine > maxFine && maxFine >= 0) || fine < 0)
 			{
 				maxFine = fine;
-				reason = "carrying illegal cargo.";
+				reason = " for carrying illegal cargo.";
+
+				for(const Mission &mission : player.Missions())
+				{
+					// Append the illegalCargoMessage from each applicable mission, if available
+					std::string illegalCargoMessage = mission.IllegalCargoMessage();
+					if(!illegalCargoMessage.empty())
+					{
+						reason = ".\n";
```

Comment:
Might be better to add ".\n\t" so the illegal cargo messages don't all need to start with a tab in order to get the paragraphs formatted correctly.